src/bom/bitcode.rs

```diff
@@ -263,9 +277,62 @@ fn build_bitcode_arguments(chan : &mut mpsc::Sender<Option<Event>>,
 {
     let mut orig_target = None;
 
+    // Determine the language of the original input file, so that the bitcode can
+    // utilize the same language, because using C++ mode to compile C will cause
+    // errors:
+    //
+    // char* greeting() { return "hello, world"; }
+    //
+    // The above is perfectly acceptable C, but if compile as C++ it yields:
+    // error: ISO C++11 does not allow conversion from string literal to 'char*'.
+    let is_c_plusplus = orig_args.iter().find(
+        |&arg| arg.to_str().map(|a| !a.starts_with("-") &&
+                                (a.ends_with(".cc") ||
+                                 a.ends_with(".cpp")))
+            .unwrap_or_else(|| false))
+        .is_some();
+    let file_ext = if is_c_plusplus { ".cc" } else { ".c" };
+
+    // If the native compiler is used for preprocessing, it will write to a
+    // temporary output file that is subsequently consumed by clang to generate
+    // the bitcode.
+    let mut preprocess = ops.push_op(
+        &run(&Executable::new(orig_compiler_cmd,
+                              ExeFileSpec::Append,
+                              ExeFileSpec::option("-o")), &None)
```

Comment:
Does this operation deserve a label?

Reply:
It automatically sets the label to be the name of the executable, and since that is a variable, that seemed like the best option here for user information.